wazimap_ng/general/widgets.py

```diff
@@ -63,10 +68,15 @@ def get_context(self, name, value, attrs=None):
             value = queryset.get(id=value)
             selected_permission = value.dataset.permission_type
 
+        profile_id = None
+        if self.instance is not None:
+            profile_id = self.instance.profile_id
+
```

Comment:
I think we can filter queryset according to profile_id
Q(indicator__dataset__permission_type="public") | Q(indicator__dataset__permision_type="private", dataset_profile_id=profile_id)

Reply:
done 👍